ironfish/src/wallet/walletdb/walletdb.ts

```diff
@@ -572,4 +584,38 @@ export class WalletDB {
   ): Promise<void> {
     await this.pendingTransactionHashes.clear(tx, account.prefixRange)
   }
+
+  async cleanupDeletedAccounts(signal?: AbortSignal): Promise<void> {
+    let recordsToCleanup = 1000
+
+    const stores: IDatabaseStore<{
+      key: Readonly<unknown>
+      value: unknown
+    }>[] = [
+      this.transactions,
+      this.sequenceToNoteHash,
+      this.nonChainNoteHashes,
+      this.nullifierToNoteHash,
+      this.pendingTransactionHashes,
+      this.decryptedNotes,
+    ]
+
+    for (const [accountId] of await this.accountIdsToCleanup.getAll()) {
+      const prefix = calculateAccountPrefix(accountId)
+      const range = StorageUtils.getPrefixKeyRange(prefix)
+
+      for (const store of stores) {
+        for await (const key of store.getAllKeysIter(undefined, range)) {
+          if (signal?.aborted === true || recordsToCleanup === 0) {
```

Comment:
I am a little confused here. Does the value of signal.aborted change during the execution of this function? 

If the value does not change,  it seems when signal is aborted, no need to call cleanuoDeletedAccounts function. Can we move the condition outside of the function instead of doing it in every for iteration?

Reply:
Yes, it can change during the execution of this function. It's an AbortedController which is used to signal to an asynchronous task that it should stop. https://developer.mozilla.org/en-US/docs/Web/API/AbortController